src/runtime/tests/arc-test.ts

```diff
@@ -1229,7 +1229,11 @@ describe('Arc storage migration', () => {
       assert.equal(worldThing0.rawData.name, 'world');
       // `world` entity was added 1s after `hello`.
       // This also verifies `hello` wasn't update when being re-added.
-      assert.isTrue(worldThing0.expirationTimestamp - helloThing0.expirationTimestamp > 1000);
+      if (worldThing0.expirationTimestamp - helloThing0.expirationTimestamp < 1000) {
+        console.warn(`Flaky test: worldThing0.expirationTimestamp - helloThing0.expirationTimestamp` +
+            `${worldThing0.expirationTimestamp} - ${helloThing0.expirationTimestamp} < 1000`);
+      }
+      assert.isTrue(worldThing0.expirationTimestamp - helloThing0.expirationTimestamp >= 1000);
```

Comment:
Unless we are using a fake/settable clock, this seems like it will always be fragile. Logic-flow-wise, where does our expectation that hello's timestamp will be >= 1000 come from? Are we relying on some implementation detail around how arc.idle or something else works, time-polling-wise?

Reply:
the particle code is supposed to wait `await new Promise(resolve => setTimeout(resolve, 1000));` before adding the `world` entity.